### PR TITLE
Use rest v1.1 for all Reader calls

### LINKED ISSUE #2359: Use v1.1 for all Reader REST calls
Right now the Android reader uses a mix of 1.0 & 1.1 calls, but it should be using only 1.1 now. This means every reader action that currently uses `getRestClientUtils()` should change to use `getRestClientUtilsV1_1()`.

### PR DESCRIPTION
**Pull request by nbradbury**
Fix #2359 - updated the Reader to use v1.1 REST endpoints for all actions.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderPostActions.java [hunk 1]
```diff
@@ -470,4 +471,27 @@ private static String getEndpointForTag(ReaderTag tag) {
...
+     * between API versions (as it did when we moved from v1 to v1.1)
+     *
+     * ex: https://public-api.wordpress.com/rest/v1/read/tags/fitness/posts
+     *     becomes just                            /read/tags/fitness/posts
+     */
+    private static String getRelativeEndpoint(final String endpoint) {
+        if (endpoint != null && endpoint.startsWith("http")) {
+            int pos = endpoint.indexOf("/read/");
+            if (pos > -1) {
+                return endpoint.substring(pos, endpoint.length());
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderPostActions.java]: Should not be ```endpoint.length()+1``` and remove the first slash?

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderPostActions.java]: I wanted to keep the slash to be consistent with commit b6bd5ba

Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderPostActions.java]: Sorry, I had read b6bd5ba in the opposite way.
Wonder why we're adding the leading slash when both V1 and V1.1 of the REST Endpoints already have it in the URL
```
protected static final String REST_API_ENDPOINT_URL_V1 = "https://public-api.wordpress.com/rest/v1/";
protected static final String REST_API_ENDPOINT_URL_V1_1 = "https://public-api.wordpress.com/rest/v1.1/";
```
and remove our first leading slash internally by calling ```getAbsoluteURL ``` https://github.com/Automattic/android-wordpress-com-rest/blob/master/WordPressComRest/src/main/java/com/wordpress/rest/RestClient.java#L69

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderPostActions.java]: Hadn't noticed we were doing that! I'll update this PR to remove the initial slash.

**Comment by nbradbury** [comment]
@daniloercoli Leading slashes are now removed, ready for re-review.

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**